src/TEKFrame.java

```diff
@@ -48,6 +54,102 @@ public TEKFrame(){
             (int)(getHeight()-getInsets().top-getInsets().bottom+10*scrollPane.getViewportBorderBounds().getHeight())));
         
         this.add(scrollPane);
+        
+         // Initialize the list of ObjectUIs
+        objects = new ArrayList<>();
+        createSampleObjects(panel); // create sample objects
+        
+        // Adding menu options for Open and Save
+        JMenuBar menuBar = new JMenuBar();
+        JMenu fileMenu = new JMenu("File");
+        
+        
+        JMenuItem openItem = new JMenuItem("Open");
+        openItem.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                openFile();
+            }
+        });
+        
+        JMenuItem saveItem = new JMenuItem("Save");
+        saveItem.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                saveFile();
+            }
+        });
+
+        fileMenu.add(openItem);
+        fileMenu.add(saveItem);
+        menuBar.add(fileMenu);
+
+        setJMenuBar(menuBar);
```

Comment:
Right below this line, add a `this.pack();`  It will update the entire frame to include the JMenuBar, as sometimes it will be visible until the user maximizes, resizes, or moves the frame.